Add concept for Package and Image Mode Hosts (#3611)
theforeman/foreman-documentation: guides/common/modules/con_understanding-package-mode-and-image-mode-hosts.adoc

[id="understanding_package_mode_and_image_mode_hosts_{context}"]
= Understanding Package mode and Image mode hosts

> In the latest update, hosts can now operate in two modes: package mode and image mode.

Reviewer: ```suggestion
You can operate hosts in two modes: package mode and image mode.
```

.Image mode hosts

> Image mode hosts include eight new fields that reflect their boot status.

Reviewer: ```suggestion
Image mode hosts include additional fields that reflect their boot status.
```

I would not use "new" because it will get outdated & maybe also drop "eight" because it might become more/less fields in the future.

These fields track the container images used in various scenarios:

* *Booted Image & Digest*: Indicates the container image the host is currently using.
* *Staged Image & Digest*: Indicates the container image the host uses after the next reboot.
* *Available Image & Digest*: Represents the container image cached by running `Bootc upgrade --check`.
* *Rollback Image & Digest*: Represents the container image the host reverts to after the next reboot if `Bootc rollback` is applied.

.All host listing page updates

> * *New Column for Host Types*: The host listing page includes a column with icons that indicate the host type.

Reviewer 2: Note: this is only on the _new_ hosts overview page, which is not the default yet.

** Clicking an icon displays the image path and digest.
** Note that the image path can be used directly with podman or docker to pull the image from a registry.
* *Search Enhancements*:
** Use `image_mode=true` or `image_mode=false` to filter hosts by type.
** Unregistered systems are assumed to be in package mode.
** Search using eight new fields:
`Bootc_booted_image`, `Bootc_booted_digest`, `Bootc_staged_image`, `Bootc_staged_digest`, `Bootc_available_image`, `Bootc_available_digest`, `Bootc_rollback_image`, `Bootc_rollback_digest`.

.Host Details Page Enhancements

> A new card appears on the Details tab for registered hosts, displaying the eight fields from `Bootc`.

Reviewer: I suggest to reword "eight" here too.

.Important Notes

> * *Pull-Mode Remote Execution*: For pull-mode remote execution users, related packages must be included in the image mode host's container image.

Reviewer: Not blocking your PR:

@ianballou Does that mean _all_ packages that are not present on a default RHEL base OS should be installed manually? I think this could also affect `subscription-manager` which ATIX packages for Debian/Ubuntu and SLES: https://oss.atix.de/ If it does, then it's probably even more packages such as OpenSCAP client and more.

Do you think that some packages such as `katello-host-tools-tracer` are not necessary anymore for container-based hosts because you'd rather rebuild+redeploy instead of updating?

Reviewer 2: `subscription-manager` at least is available by default for Enterprise Linux systems, but you're right that any packages the user may need should get baked into the image. Others can be installed transiently after running `bootc usr-overlay`, but that is generally not recommended since it creates drift.

`subscription-manager` is crucial for the image mode feature since it is responsible for sending the bootc facts.

You're right about Tracer not being very useful for image mode hosts unless you are using the `bootc usr-overlay`.

> [id="understanding_package_mode_and_image_mode_hosts_{context}"]

Reviewer: ```suggestion
[id="understanding-package-mode-and-image-mode-hosts"]
```

suggestion (non-blocking): Align the anchor with the file name & drop context. Using the context attribute often makes it harder to maintain links. If anchor and file name use dashes, then it's ever easier to find things in the repoitory.

> = Understanding Package mode and Image mode hosts

Reviewer 3: Concept heading shouldn§t start with a gerundium.

* *Staged Image & Digest*: Indicates the container image the host uses after the next reboot.
* *Available Image & Digest*: Represents the container image cached by running `bootc upgrade --check`.
* *Rollback Image & Digest*: Represents the container image the host reverts to after the next reboot if `Bootc rollback` is applied.

.All host listing page updates
* *New Column for Host Types*: On the new Hosts Overview page, the host listing page includes a column with icons that indicate the host type.
Note that users can add or remove the image mode column by using the *Manage columns* feature.
** Clicking an icon displays the image path and digest.
** Note that the image path can be used directly with podman or docker to pull the image from a registry.
* *Search Enhancements*:
** Use `image_mode=true` or `image_mode=false` to filter hosts by type.
** Unregistered systems are assumed to be in package mode.
** Search using eight new fields:
`Bootc_booted_image`, `Bootc_booted_digest`, `Bootc_staged_image`, `Bootc_staged_digest`, `Bootc_available_image`, `Bootc_available_digest`, `Bootc_rollback_image`, `Bootc_rollback_digest`.

.Host Details Page Enhancements

> A new card appears on the Details tab for registered hosts, displaying the additional fields from `Bootc`.

Reviewer 3: This doesn't belong here - we don't describe web UI features like this. 
I think this could go to release notes instead but I'd drop it here.

.Important Notes

> * *Pull-Mode Remote Execution*: For pull-mode remote execution users, related packages must be included in the image mode host's container image.

Reviewer 3: Perhaps this should be a prerequisite in the procedure? (I'm not sure because it isn't clear to me why that's needed.)

> This distinction between package mode and image mode provides greater flexibility in managing host systems and streamlines operations for image-based deployments.

Reviewer: ```suggestion
This distinction between package mode and image mode provides greater flexibility in managing hosts and streamlines operations for image-based deployments.
```

suggestion (non-blocking): Align with our default wording "hosts".
